drivers/storage/portworx/component/disruption_budget.go

```diff
@@ -0,0 +1,191 @@
+	}
+
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	if err := c.createKVDBPodDisruptionBudget(cluster, ownerRef); err != nil {
+		return err
+	}
+	if err := c.createPortworxPodDisruptionBudget(cluster, ownerRef); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *disruptionBudget) Delete(cluster *corev1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	if err := k8sutil.DeletePodDisruptionBudget(
+		c.k8sClient, KVDBPodDisruptionBudgetName,
+		cluster.Namespace, *ownerRef,
+	); err != nil {
+		return err
+	}
+	if err := k8sutil.DeletePodDisruptionBudget(
+		c.k8sClient, StoragePodDisruptionBudgetName,
+		cluster.Namespace, *ownerRef,
+	); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *disruptionBudget) MarkDeleted() {}
+
+func (c *disruptionBudget) createPortworxPodDisruptionBudget(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	var err error
+	c.sdkConn, err = pxutil.GetPortworxConn(c.sdkConn, c.k8sClient, cluster.Namespace)
+	if err != nil {
+		return err
+	}
+
+	nodeClient := api.NewOpenStorageNodeClient(c.sdkConn)
+	ctx, err := pxutil.SetupContextWithToken(context.Background(), cluster, c.k8sClient)
+	if err != nil {
+		c.closeSdkConn()
+		return err
+	}
+
+	nodeEnumerateResponse, err := nodeClient.EnumerateWithFilters(
+		ctx,
+		&api.SdkNodeEnumerateWithFiltersRequest{},
+	)
+	if err != nil {
+		c.closeSdkConn()
+		return fmt.Errorf("failed to enumerate nodes: %v", err)
+	}
+
+	storageNodesCount := 0
+	for _, node := range nodeEnumerateResponse.Nodes {
+		if len(node.Pools) > 0 && node.Pools[0] != nil {
+			storageNodesCount++
+		}
+	}
+
+	// Create PDB only if there are at least 3 nodes. With 2 nodes are less, if 1
+	// node goes down Portworx will lose quorum anyway. Such clusters would be
+	// non-prod clusters and there is no point in blocking the evictions.
+	if storageNodesCount > 2 {
+		minAvailable := intstr.FromInt(storageNodesCount - 1)
+		pdb := &policyv1beta1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            StoragePodDisruptionBudgetName,
+				Namespace:       cluster.Namespace,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+			Spec: policyv1beta1.PodDisruptionBudgetSpec{
+				MinAvailable: &minAvailable,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						constants.LabelKeyClusterName: cluster.Name,
+						constants.LabelKeyStoragePod:  constants.LabelValueTrue,
+					},
+				},
+			},
+		}
+		return k8sutil.CreateOrUpdatePodDisruptionBudget(c.k8sClient, pdb, ownerRef)
+	}
+	return nil
+}
+
+func (c *disruptionBudget) createKVDBPodDisruptionBudget(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	// No need to create PDB for KVDB when using external KVDB
+	if cluster.Spec.Kvdb != nil && !cluster.Spec.Kvdb.Internal {
+		return nil
+	}
+
+	minAvailable := intstr.FromInt(2)
```

Comment:
We do support kvdb cluster size of 5. We can check from the StorageCluster object if the "--kvdb_cluster_size" argument is provided and then decide whether to use 2 or 3.

Reply:
You mean check the argument from misc-args?

Reply:
Yes